interfaces/builtin/mir.go

```diff
@@ -74,6 +74,7 @@ var mirConnectedPlugAppArmor = []byte(`
 unix (receive, send) type=seqpacket addr=none peer=(label=###SLOT_SECURITY_TAGS###),
 /run/mir_socket rw,
 /run/user/[0-9]*/mir_socket rw,
+/run/user/[0-9]*/snap.###SLOT_NAME###/mir_socket rw,
```

Comment:
So this is an interesting issue.  Unity8 by default just puts the socket in XDG_RUNTIME_DIR, which is how that path ended up being what it is.  That's certainly changeable to be hardcoded instead.

However, this does not necessarily create a tight coupling to our slot implementation.  That socket path is exported to apps via an environment variable, passed by ubuntu-app-launch I believe.  So there's no hard-coding on the plug side.  (Although it does mean we rely on our slot's implementation of ubuntu-app-launch which probably isn't good.)

So because of needing UAL, and because my approach does leak the name of the slot, it makes sense to use the global path instead.

There is one downside to that though, which is that we can't have nested Mir servers that way.  This mir interface allows a top-level (/run/mir_socket), one nested one (user socket) and no more.  If we allowed each slot to have its own socket, then we can be a little future proof in terms of what we want to allow with Mir servers.

But unless I hear a change of heart, I'll go ahead with your recommendation.